ci/test.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+set -x
+
+this_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+proj_dir=${this_dir}/../
+
+# setup temp dirs
+build_dir=$(mktemp -d)
+trap "rm -rf ${build_dir}" EXIT
+
+pushd ${build_dir}
```

Comment:
btw, this can just be a subshell via parenthesis. i.e.: 

`(cmake ${proj_dir && make -j $(nproc) VERBOSE=1 )` and then you don't need popd